src/controllers/authController.js

```diff
@@ -0,0 +1,188 @@
+    accessToken,
+  });
+};
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validatePassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+
+  await userService.register(name, email, hashedPass);
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = await bcrypt.compare(password, user.password);
+
+  if (!isPasswordValid) {
+    throw ApiError.badRequest('Wrong password');
+  }
+
+  generateTokens(res, user);
+};
+
+const refresh = async (req, res) => {
+  const { refreshToken } = req.cookies;
+
+  const userData = await jwtService.verifyRefresh(refreshToken);
+  const token = await tokenService.getByToken(refreshToken);
+
+  if (!userData || !token) {
+    throw ApiError.unauthorized();
+  }
+
+  const user = await userService.findByEmail(userData.email);
+
+  generateTokens(res, user);
+};
+
+const logout = async (req, res) => {
+  const { refreshToken } = req.cookies;
+  const userData = await jwtService.verifyRefresh(refreshToken);
+
+  if (!userData || !refreshToken) {
+    throw ApiError.unauthorized();
+  }
+
+  await tokenService.remove(userData.id);
+  res.sendStatus(204);
+};
+
+const sendResetPassword = async (req, res) => {
+  const { email } = req.body;
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const resetToken = uuidv4();
+
+  user.resetToken = resetToken;
+  await user.save();
+
+  await emailService.sendResetPasswordEmail(email, resetToken);
+
+  res.sendStatus(200);
+};
+
+const resetPassword = async (req, res) => {
+  const { resetToken } = req.params;
+  const { password, email } = req.body;
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  if (user.resetToken !== resetToken) {
+    throw ApiError.badRequest('Invalid or expired reset token');
+  }
+
+  const isMatch = await bcrypt.compare(password, user.password);
+
+  if (isMatch) {
+    throw ApiError.badRequest(
+      'The new password must be different from the current one',
+    );
+  }
+
+  const hashedPassword = await bcrypt.hash(password, 10);
+
+  await userService.updatePassword(user.id, hashedPassword);
+
+  user.resetToken = null;
+  await user.save();
```

Comment:
Ensure that `user.save()` is awaited to handle any potential errors during the save operation.